### DISCUSSION TITLE
Questions about development a form

### DISCUSSION THREAD

**Discussion by Kota-M** [craftcms/contact-form#230] [Q&A]
Hello, nice to meet you.
I have been implementing my form with the plugin, and would like to ask some "how to" questions.

1) I read the section of "The afterValidate event", but even if I put the code in config file, the validation does not work.
Where should I put the validation code?

2) My auto-respond email is like this;

* Name: Kota
* Email: kota@.....
* Phone: 647-xxx-xxxx
* fileNumber: 1234567

Is it possible to replace "fileNumber" with "File Number"?
It is from the field ID, but it seems not possible to contain a space.

**Comment by AugustMiller** [comment]
1. Event handlers belong in a custom module or plugin. See the [documentation on events](https://craftcms.com/docs/5.x/extend/events.html) for more info!
2. You can provide labels for additional message sub-fields by adding a [static translation file](https://craftcms.com/docs/5.x/system/sites.html#static-message-translations) to your project. The key would be your field name (`fileNumber`) and the value would be your human-facing label (`File Number`).
